File: FinBalancer.Api/Models/Achievement.cs
namespace FinBalancer.Api.Models;

public class Achievement
{
    public Guid Id { get; set; }
    public string Key { get; set; } = string.Empty; // first_transaction, goal_reached, etc.
    public string Name { get; set; } = string.Empty;
    public string Icon { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public DateTime? UnlockedAt { get; set; }
}

File: FinBalancer.App/lib/models/achievement.dart
class Achievement {
  final String id;
  final String key;
  final String name;
  final String icon;
  final String description;
  final String period; // daily, monthly, yearly, weekly, lifetime
  final DateTime? unlockedAt;

  Achievement({
    required this.id,
    required this.key,
    required this.name,
    required this.icon,
    required this.description,
    this.period = 'lifetime',
    this.unlockedAt,
  });

  bool get isUnlocked => unlockedAt != null;

  String get periodLabel {
    switch (period) {
      case 'daily': return 'Daily';
      case 'weekly': return 'Weekly';
      case 'monthly': return 'Monthly';
      case 'yearly': return 'Yearly';
      default: return 'Lifetime';
    }
  }

  factory Achievement.fromJson(Map<String, dynamic> json) {
    return Achievement(
      id: json['id']?.toString() ?? '',
      key: json['key'] as String? ?? '',
      name: json['name'] as String? ?? '',
      icon: json['icon'] as String? ?? '',
      description: json['description'] as String? ?? '',
      period: json['period'] as String? ?? 'lifetime',
      unlockedAt: json['unlockedAt'] != null ? DateTime.tryParse(json['unlockedAt'] as String) : null,
    );
  }
}
